### PR TITLE
qvim, vim_configurable: get rid of composableDerivation

### PR DESCRIPTION
**Pull request by FRidh**
Use `stdenv.mkDerivation` directly instead of `composableDerivation`.

Some configure flags may have changed as the conversion wasn't exactly
straightforward.

---

Please test this change. I am no user of this package. Will merge in a week from now.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
cc contributors @Ma27 @LnL7 @bjornfor @7c6f434c @vcunat

**Comment by Mic92** [comment]
Then only qvim is left that uses this obscure function.

**Comment by Mic92** [comment]
@GrahamcOfBorg build vimNox ezquake vim_configurable vimHugeX vimPlugins.pluginnames2nix

**Comment by FRidh** [comment]
@Mic92 don't forget `php`. That one is big...

**Comment by LnL7** [comment]
I thought php and vim_configurable where the only remaining.

**Comment by FRidh** [comment]
I've also added `qvim` to this PR. cc @ttuegel @grwlf

The expressions are quite similar, so it would also be good to afterwards get them more in line with each other.

File: pkgs/applications/editors/vim/qvim.nix [hunk 1]
```diff
@@ -1,110 +1,94 @@
...
-composableDerivation {
-} (fix: {
+  name = "qvim-7.4." + tag;
 
-    name = "qvim-7.4." + tag;
+  enableParallelBuilding = true; # test this
 
-    enableParallelBuilding = true; # test this
-
-    src = fetchgit {
```
Comment (LnL7) [inline pkgs/applications/editors/vim/qvim.nix]: Is this actually different from vim_configurable? It looks like a copy-paste with a different source and that's already configurable (eg. nox)

**Pull request merged**